modules/code-deployment/src/test/java/org/apache/ignite/deployment/metastore/DeploymentUnitStoreImplTest.java

```diff
@@ -58,7 +58,7 @@ public class DeploymentUnitStoreImplTest {
 
     private final VaultManager vaultManager = new VaultManager(new InMemoryVaultService());
 
-    private final List<UnitNodeStatus> history = new ArrayList<>();
+    private final Set<UnitNodeStatus> history = ConcurrentHashMap.newKeySet();
```

Comment:
Set is not the best solution here. It might hide the duplicates. I think duplicates are an important part to check here.